docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,302 @@
+pick an instance type that has sufficient instance storage, unless you can make
+do with whatever space is left on your root partition.
+
+Note: The master uses a persistent volume ([etcd](architecture.md#etcd)) to track
+its state but similar to the nodes, containers are mostly run against instance
+storage, except that we repoint some important data onto the peristent volume.
+
+The default storage driver for Docker images is aufs.  Specifying btrfs (by passing the environment
+variable `DOCKER_STORAGE=btrfs` to kube-up) is also a good choice for a filesystem.  btrfs
+is relatively reliable with Docker and has improved its reliability with modern
+kernels.  It can easily span multiple volumes, which is particularly useful
+when we are using an instance type with multiple ephemeral instance disks.
+
+### AutoScaling
+
+Nodes (but not the master) are run in an
+[AutoScalingGroup](http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/AutoScalingGroup.html)
+on AWS.  Currently auto-scaling (e.g. based on CPU) is not actually enabled
+([#11935](http://issues.k8s.io/11935)).  Instead, the auto-scaling group means
+that AWS will relaunch any nodes that are terminated.
+
+We do not currently run the master in an AutoScalingGroup, but we should
+([#11934](http://issues.k8s.io/11934)).
+
+### Networking
+
+Kubernetes uses an IP-per-pod model.  This means that a node, which runs many
+pods, must have many IPs.  AWS uses virtual private clouds (VPCs) and advanced
+routing support so each pod is assigned a /24 CIDR.  Each pod is assigned a /24
+CIDR; the assigned CIDR is then configured to route to an instance in the VPC
+routing table.
+
+It is also possible to use overlay networking on AWS, but that is not the
+configuration of the kube-up script.
+
+### NodePort and LoadBalancing
+
+Kubernetes on AWS integrates with [Elastic Load Balancing
+(ELB)](http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/US_SetUpASLBApp.html).
+When you create a service with `Type=LoadBalancer`, Kubernetes (the
+kube-controller-manager) will create an ELB, create a security group for the
+ELB which allows access on the service ports, attach all the nodes to the ELB,
+and modify the security group for the nodes to allow traffic from the ELB to
+the nodes.  This traffic reaches kube-proxy where it is then forwarded to the
+pods.
+
+ELB has some restrictions: it requires that all nodes listen on a single port,
+and it acts as a forwarding proxy (i.e. the source IP is not preserved).  To
+work with these restrictions, in Kubernetes, [LoadBalancer
+services](../user-guide/services.html#type-loadbalancer) are exposed as
+[NodePort services](../user-guide/services.md#type-nodeport).  Then
+kube-proxy listens externally on the cluster-wide port that's assigned to
+NodePort services and forwards traffic to the corresponding pods.  So ELB is
+configured to proxy traffic on the public port (e.g. port 80) to the NodePort
```

Comment:
Let me try to make this example clearer. Can you please make  a new paragraph and revise to:

So for example, if ELB is configured to proxy traffic on the public port 80 to the NodePort that is assigned to the service 31234. Then any in-coming traffic that is sent to that NodePort (e.g. port 31234) is recognized by kube-proxy and sent to the correct pods for that service.